source/plugins/go/src/oms.go

```diff
@@ -1116,6 +1119,146 @@ func UpdateNumTelegrafMetricsSentTelemetry(numMetricsSent int, numSendErrors int
 	ContainerLogTelemetryMutex.Unlock()
 }
 
+func processIncludes(kubernetesMetadataMap map[string]interface{}, includesList []string) map[string]interface{} {
+	includedMetadata := make(map[string]interface{})
+
+	// pre process image related fields
+	var imageRepo, imageName, imageTag, imageID string
+	imageProcessed := false // Flag to check if image processing is required
+	for _, include := range includesList {
+		if include == "imageid" || include == "imagerepo" || include == "image" || include == "imagetag" {
+			imageProcessed = true
+			break
+		}
+	}
+
+	if imageProcessed {
+		if hash, ok := kubernetesMetadataMap["container_hash"].(string); ok {
+			imageID = extractImageID(hash)
+		}
+		if image, ok := kubernetesMetadataMap["container_image"].(string); ok {
+			imageRepo, imageName, imageTag = parseImageDetails(image)
+		}
+	}
+
+	for _, include := range includesList {
+		switch include {
+		case "poduid":
+			if val, ok := kubernetesMetadataMap["pod_id"]; ok {
+				includedMetadata["podUid"] = val
+			}
+		case "podlabels":
+			if val, ok := kubernetesMetadataMap["labels"]; ok {
+				includedMetadata["podLabels"] = val
+			}
+		case "podannotations":
+			if val, ok := kubernetesMetadataMap["annotations"]; ok {
+				if annotationsMap, ok := val.(map[string]interface{}); ok {
+					filteredAnnotations := make(map[string]interface{})
+					for key, annotationValue := range annotationsMap {
+						if !strings.Contains(key, "kubernetes.io/config") {
+							filteredAnnotations[key] = annotationValue
+						}
+					}
+					includedMetadata["podAnnotations"] = filteredAnnotations
+				}
+			}
+		case "imageid":
+			if imageID != "" {
+				includedMetadata["imageID"] = imageID
+			}
+		case "imagerepo":
+			if imageRepo != "" {
+				includedMetadata["imageRepo"] = imageRepo
+			}
+		case "image":
+			if imageName != "" {
+				includedMetadata["image"] = imageName
+			}
+		case "imagetag":
+			if imageTag != "" {
+				includedMetadata["imageTag"] = imageTag
+			}
+		}
+	}
+	return includedMetadata
+}
+
+func extractImageID(hash string) string {
+	if atLocation := strings.Index(hash, "@"); atLocation != -1 {
+		return hash[atLocation+1:]
+	}
+	return ""
+}
+
+func parseImageDetails(image string) (repo, name, tag string) {
+	slashLocation := strings.Index(image, "/")
+	colonLocation := strings.Index(image, ":")
+	atLocation := strings.Index(image, "@")
+
+	// Exclude the digest part for imageRepo/image/tag parsing, if present
+	if atLocation != -1 {
+		image = image[:atLocation]
+	}
+
+	// Process Image Name, Repo, and Tag based on the original logic
+	if colonLocation != -1 {
```

Comment:
Can't we just make colonLocation as the end of the list if it is not found? to avoid all the duplicate code?

Reply:
changed